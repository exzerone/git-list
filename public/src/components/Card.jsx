import React from 'react';

function Card(props){
	const { user } = props;

	return (
		<div className="user_card">
			<img alt="avatar" src={user.avatar_url} style={{height: '150px', width: '150px'}}/>
			<div>
				<h3>User: {user.login}</h3>
				<div>Site Admin: {user['site_admin'] ? 'Yes': 'No'} </div>
			</div>
		</div>
	);
}

export default Card;

