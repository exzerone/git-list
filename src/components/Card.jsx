import React from 'react';
import styled from 'styled-components';

const UserCard = styled.div`
	border: 1px solid black;
	height: 350px;
	width: 300px;
	margin: 5px;
`;

const UserTitle = styled.h3`
	max-width: 300px;
	height: 50px;
	background-color: lightgray;
	margin: 0;
	padding: 10px;
`;

const UserDesc = styled.div`
	padding: 10px;
`;

function Card({user}){
	return (
		<UserCard className="user_card">
			<UserTitle className="user_title">username: {user.login}</UserTitle>
			<UserDesc className="user_description">
				<img alt="avatar" src={user.avatar_url} style={{height: '175px', width: '225px'}}/>
				<div>
					<div>Github Url: {user.html_url}</div>
					<div>Site Admin: {user.site_admin ? 'Yes': 'No'} </div>
				</div>
			</UserDesc>
		</UserCard>
	);
}

export default Card;

