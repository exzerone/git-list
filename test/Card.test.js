import React from 'react';
import { shallow } from 'enzyme';
import Card from '../src/components/Card.jsx';
const data = require('../data.json');

let wrapper = shallow(<Card user={data[0]}></Card>);

describe("Card", ()=> {
	it('should render without crashing', ()=>{
		expect(wrapper).toMatchSnapshot();
	});

	it('renders the user title', ()=>{
		expect(wrapper.find('.user_title').text()).toEqual(`${data[0]['login']}`);
	});
});

