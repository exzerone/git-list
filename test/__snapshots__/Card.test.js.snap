// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render without crashing 1`] = `
<styled.div
  className="user_card"
>
  <styled.h3
    className="user_title"
  >
    jvantuyl
  </styled.h3>
  <styled.div
    className="user_description"
  >
    <img
      alt="avatar"
      src="https://avatars0.githubusercontent.com/u/101?v=4"
      style={
        Object {
          "borderRadius": "15px",
          "height": "150px",
          "width": "150px",
        }
      }
    />
  </styled.div>
</styled.div>
`;
